app/models/concerns/visible.rb

```diff
@@ -0,0 +1,24 @@
+module Visible
+
+  #Validations are methods called at the class level
+  #this extend give us a simpler way to include them.
+
+  extend ActiveSupport::Concern
+
+  VALID_STATUSES = ['public','private','archived']
+  
+  def archived?
+    status == 'archived'
+  end
+
+  included do
+    validates :status, inclusion: { in: VALID_STATUSES }
+  end
+
+  class_methods do
+    def public_count
+      where(status: 'public').count
+    end
+  end
+
+end
```

Comment:
@JuanMaGiraldo the closing blank space is missing in this file.

Reply:
Sorry, I made the first pull request without the Visual Studio configuration.